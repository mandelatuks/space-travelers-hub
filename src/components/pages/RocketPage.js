import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { FetchRockets } from '../../Redux/Rocket/rockets';
import Rocket from '../Rocket';

const RocketPage = () => {
  const rockets = useSelector((state) => state.rocketsReducer);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(FetchRockets());
  }, [dispatch]);

  const data = rockets.map((rocket) => (
    <Rocket key={rocket.rocket_id} rocket={rocket} />
  ));
  return <div className="rocketList">{data}</div>;
};

export default RocketPage;
