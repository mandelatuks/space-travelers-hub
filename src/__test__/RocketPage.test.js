import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../Redux/Configurestore';
import RocketPage from '../components/pages/RocketPage';

describe('it renders on the screen:', () => {
  test('it renders Rockets correctly', () => {
    render(
      <Provider store={store}>
        <RocketPage />
      </Provider>,
    );
    expect(screen.queryByText(/Missions/)).toBeNull();
    expect(screen.queryByText(/Description/)).toBeNull();
    expect(screen.queryByText(/Status/)).toBeNull();
  });

  test('it displays snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <RocketPage />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
