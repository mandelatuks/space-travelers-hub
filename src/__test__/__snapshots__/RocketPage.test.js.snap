// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders on the screen: it displays snapshot 1`] = `null`;
